lib/kino.ex

```diff
@@ -289,17 +289,21 @@ defmodule Kino do
   def animate(interval_ms, state, fun) when is_integer(interval_ms) and is_function(fun, 2) do
     frame = Kino.Frame.new()
 
-    Kino.Frame.periodically(frame, interval_ms, {0, state}, fn {i, state} ->
+    stream =
+      Stream.interval(interval_ms)
+      |> Stream.take(interval_ms)
```

Comment:
I see. But, we would still need to combine with `Stream.take/1`, wouldn't we? Otherwise, the iterations won't stop. 😓

![image (2)](https://user-images.githubusercontent.com/384201/234932344-4f135b44-fe85-4d37-8553-8f56edc93e07.png)

Reply:
It is up to the user, we have `animate/3` with reducer and any iteration can `:halt` (same as `periodically`). Another way is to pass a non-infinite stream.

Reply:
Got it. Thanks for the explanation. ❤️